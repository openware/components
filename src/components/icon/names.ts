const stringLitArray = <L extends string>(arr: L[]) => arr;
export const OwIconNameList = stringLitArray([
    'apps',
    'arrow-back',
    'arrow-back-ios',
    'arrow-downward',
    'arrow-drop-down',
    'arrow-drop-up',
    'arrow-forward',
    'arrow-forward-ios',
    'arrow-left',
    'arrow-right',
    'arrow-upward',
    'bar-chart',
    'calendar',
    'close',
    'close-circle',
    'dashboard',
    'done',
    'download',
    'exit-to-app',
    'facebook',
    'facebook-circle',
    'help-circle',
    'help-outline',
    'history',
    'keyboard-arrow-down',
    'keyboard-arrow-left',
    'keyboard-arrow-right',
    'keyboard-arrow-up',
    'linkedin',
    'linkedin-circle',
    'lock',
    'lock-open-outline',
    'lock-outline',
    'logout',
    'menu',
    'person',
    'person-outline',
    'pie-chart',
    'reddit',
    'reddit-circle',
    'search',
    'security',
    'settings',
    'share',
    'shield',
    'show-chart',
    'sun',
    'telegram',
    'telegram-circle',
    'time',
    'twitter',
    'twitter-circle',
    'wallet',
    'youtube',
    'youtube-circle',
    'moon',
]);

export type OwIconName = typeof OwIconNameList[number];
