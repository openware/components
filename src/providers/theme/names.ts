export type ThemeName = 'light' | 'dark' | string;
