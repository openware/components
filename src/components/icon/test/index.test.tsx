import React from 'react';
import { mount, ReactWrapper, shallow } from 'enzyme';

import { OwIcon, OwIconProps } from '../index';
import { OwIconName } from '../names';

const defaultProps: OwIconProps = {
    name: 'apps',
};

const setup = (props: Partial<OwIconProps> = {}) => {
    return shallow(<OwIcon {...defaultProps} {...props} />);
};

const setupMount = (props: Partial<OwIconProps> = {}) => {
    return mount(<OwIcon {...defaultProps} {...props} />);
};

describe('Button', () => {
    it('render', () => {
        expect(setup()).toMatchSnapshot();
    });

    it('render styled', () => {
        expect(
            setup({ style: { height: 30, color: 'white' } })
        ).toMatchSnapshot();
    });

    it('render diff name', () => {
        expect(setup({ name: 'calendar' }).render()).toMatchSnapshot();
    });

    it('render null', () => {
        expect(setup({ name: null }).render()).toMatchSnapshot();
    });

    function checkName(component: ReactWrapper, name: OwIconName) {
        component.setProps({ name }, () => {
            expect(component.prop('name')).toEqual(name);
        });
    }

    it('name', () => {
        const component = setupMount({});
        checkName(component, 'apps');
        checkName(component, 'arrow-back-ios');
        checkName(component, 'close');
        checkName(component, 'lock');
        checkName(component, 'menu');
        checkName(component, 'settings');
        checkName(component, 'shield');
        checkName(component, 'show-chart');
    });

    it('className', () => {
        const customClassName = 'custom-class-123456';
        const component = setupMount({ className: customClassName });
        expect(component.prop('className')).toEqual(customClassName);

        const newClassName = 'new-class';
        component.setProps({ className: newClassName }, () => {
            expect(component.prop('className')).toEqual(newClassName);
        });
    });
});
