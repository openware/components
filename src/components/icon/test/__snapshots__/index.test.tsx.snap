// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<div
  className="ow-icon"
  fill="currentColor"
/>
`;

exports[`Button render diff name 1`] = `
<div
  class="ow-icon"
  fill="currentColor"
/>
`;

exports[`Button render null 1`] = `null`;

exports[`Button render styled 1`] = `
<div
  className="ow-icon"
  fill="currentColor"
  style={
    Object {
      "color": "white",
      "height": 30,
    }
  }
/>
`;
